File: src/main/java/com/greentree/example/telegram/state/ChatState.java
package com.greentree.example.telegram.state;

import org.telegram.telegrambots.meta.api.methods.send.SendMessage;
import org.telegram.telegrambots.meta.api.objects.CallbackQuery;
import org.telegram.telegrambots.meta.api.objects.Message;
import org.telegram.telegrambots.meta.bots.AbsSender;
import org.telegram.telegrambots.meta.exceptions.TelegramApiException;

public interface ChatState {

    default ChatState onMessage(AbsSender sender, Message message) throws TelegramApiException {
        send(sender, message.getChatId(), "error onMessage");
        return null;
    }

    static void send(AbsSender sender, long chatId, String text) throws TelegramApiException {
        var m = new SendMessage();
        m.setChatId(chatId);
        m.setText(text);
        sender.execute(m);
    }

    default ChatState onCallback(AbsSender sender, CallbackQuery query) throws TelegramApiException {
        send(sender, query.getMessage().getChatId(), "error onCallback");
        return null;
    }

    void init(AbsSender sender, long chatId) throws TelegramApiException;

}

File: src/main/java/com/greentree/telegram/queue/state/ChatState.kt
package com.greentree.telegram.queue.state

import com.greentree.telegram.queue.lib.redirect
import org.telegram.telegrambots.meta.api.methods.send.SendMessage
import org.telegram.telegrambots.meta.api.objects.CallbackQuery
import org.telegram.telegrambots.meta.api.objects.Message
import org.telegram.telegrambots.meta.bots.AbsSender

fun interface ChatState : StateProvider.Response {

	fun onMessage(sender: AbsSender, message: Message) {
		sender.send(message.chatId, "error onMessage")
		redirect("begin")
	}

	fun onCallback(sender: AbsSender, query: CallbackQuery) {
		sender.send(query.message.chatId, "error onCallback")
		redirect("begin")
	}

	fun init(sender: ChatSender)
}

fun AbsSender.send(chatId: Long, text: String): Message? {
	val message = SendMessage()
	message.setChatId(chatId)
	message.text = text
	return execute(message)
}